NEW: Astronomer Certified Versioning + Support Policy (#444)
astronomer/docs: cloud/stable/06_astronomer-certified/03_ac-versioning-and-support-policy.md

This document provides information on the following:
- How Astronomer Certified is versioned.
- Which versions of Astronomer Certified are currently available.
- The support schedule and end-of-support date for all versions.

For guidelines on how to upgrade from one version of AC to another, read [Upgrade Apache Airflow on Astronomer](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions).

## Versioning Scheme

An Astronomer Certified (AC) Docker image will be published for every major and minor version of Apache Airflow. For example, AC images correspond with Apache Airflow 2.0, 2.1, 2.2 etc. as they're released in the open-source project.

Astronomer Certified Docker images come in two variants:

- quay.io/astronomer/ap-airflow:<version>-buster-onbuild
- quay.io/astronomer/ap-airflow:<version>-buster

Astronomer Certified 2.1.0 with Airflow 2.1.0, for example, would be:

- `quay.io/astronomer/ap-airflow:2.1.0-buster`
- `quay.io/astronomer/ap-airflow:2.1.0-buster-onbuild`

> For the smoothest, out-of-the-box Airflow experience, we strongly recommend and default to `buster-onbuild` images in your project's `Dockerfile`. These images incorporate Docker ONBUILD commands to copy your Airflow project directory (i.e. your `packages.txt`, `requirements.txt`, `dags`, etc.) and scaffold it so you can more easily pass those files to the containers running each core Airflow component (Scheduler, Webserver, Postgres Database, Workers).

Reviewer: ```suggestion
For the smoothest, out-of-the-box Airflow experience, we strongly recommend and default to `buster-onbuild` images in your project's `Dockerfile`. These images incorporate Docker ONBUILD commands to copy your Airflow project directory and scaffold it so you can more easily pass those files to the containers running each core Airflow component.
```

Reviewer: Doesn't seem quite like the place to add this level of detail

Reviewer: Bumping this

> For complex use cases that require that you customize the AC base image, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image).

Reviewer: ```suggestion
For complex use cases that require customizing the base AC image, read [Customize your Airflow Image on Astronomer](https://www.astronomer.io/docs/cloud/stable/develop/customize-image).
```

### Hotfix Versions

> In addition to supporting the latest versions of open source Airflow on Astronomer Certified (AC), our team regularly ships bug and security hotfixes to AC images as hotfix releases.

Reviewer: ```suggestion
In addition to supporting the latest versions of open source Airflow on AC, Astronomer regularly ships bug and security fixes to AC images in hotfix releases.
```

Since H3 is a continuation of h2 and can't be linked out, we don't need to reintroduce the acronym here.

> For example, Astronomer Certified 1.10.10 has been enhanced with 4 additional patches since its initial release:

Reviewer: ```suggestion
The format for AC hotfix versions is `<base-version>-<hotfix-version>. For example, the second hotfix release for AC 2.1.0 was `2.1.0-2`.
```

Reviewer: If you take this suggestion, remove the bullets below.

Reviewer: Bumping

- 2.1.0-1
- 2.1.0-2

> - 2.1.0-3 etc.

Reviewer 2: this talks about 1.10.10, but lists hotfixes for 2.1.0

> All hotfix releases are listed in a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md), which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported by our team and made available prior to the subsequent open source release.

Reviewer: ```suggestion
All hotfix releases have a [corresponding changelog](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md) which specifies the date the hotfix was released and all individual changes made to it. Bugs that are reported by the wider Airflow community are often backported  and made available prior to the subsequent open source release.
```
Is it still "backported" if we get them before the official release?

Reviewer: Bumping for "backported" question

> For information on how to upgrade to latest hotfix release, read [Upgrade to an AC Patch Version](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions#patch-versions-of-astronomer-certified).

Reviewer 2: upgrade to >the< latest

## Available Versions

> Below you'll find a matrix of all Astronomer Certified images supported on Astronomer Cloud. For our platform's full collection of Docker images, reference [Astronomer on Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags). For more information on Alpine and Debian as distinct system distributions, read [Migrate from Alpine to Debian](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions#migrate-from-alpine-to-debian).

Reviewer: ```suggestion
The following table contains all Astronomer Certified images supported on Astronomer Cloud. For our platform's full collection of Docker images, reference [Astronomer on Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags). For more information on Alpine and Debian as distinct system distributions, read [Migrate from Alpine to Debian](https://www.astronomer.io/docs/cloud/stable/customize-airflow/manage-airflow-versions#migrate-from-alpine-to-debian).
```

| [1.10.14](https://github.com/astronomer/ap-airflow/blob/master/1.10.14/CHANGELOG.md) | FROM quay.io/astronomer/ap-airflow:1.10.14-buster-onbuild | N/A                                                           |
| [1.10.15](https://github.com/astronomer/ap-airflow/blob/master/1.10.15/CHANGELOG.md) | FROM quay.io/astronomer/ap-airflow:1.10.15-buster-onbuild | N/A                                                           |
| [2.0.0](https://github.com/astronomer/ap-airflow/blob/master/2.0.0/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.0.0-buster-onbuild   | N/A                                                           |
| [2.0.2](https://github.com/astronomer/ap-airflow/blob/master/2.0.2/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.0.2-buster-onbuild   | N/A                                                           |
| [2.1.0](https://github.com/astronomer/ap-airflow/blob/master/2.1.0/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.1.0-buster-onbuild   | N/A                                                           |

> | [2.1.1](https://github.com/astronomer/ap-airflow/blob/master/2.1.1/CHANGELOG.md)     | FROM quay.io/astronomer/ap-airflow:2.1.1-buster-onbuild   | N/A                                                           |

Reviewer 2: 2.1.3 was released recently: https://github.com/astronomer/ap-airflow/blob/master/2.1.3/CHANGELOG.md

## Support Policy

Astronomer is committed to supporting all patch versions of Astronomer Certified (AC) for a total of 18 months from the release date of the Docker image and Python wheel. Within the support window of each Airflow + Astronomer Certified version, the following is true:

> - Astronomer Certified Docker images and Python wheel for that version are available for first-time download via [Quay.io](http://quay.io), PyPi and [astronomer.io/downloads](https://support.astronomer.io).

Reviewer: ```suggestion
- AC Docker images and Python wheel for a version are available for first-time download via [Quay.io](http://quay.io), PyPi and [astronomer.io/downloads](https://support.astronomer.io).
```
What does "first-time" mean here?

Author: Addressed here - lmk what you think, agreed that "first time download" is a bit redundant

- Astronomer will regularly publish hotfixes for bug or security issues identified as high priority.
- The Astronomer Support team will offer support for paying customers running a supported version of AC via the [Astronomer Support Portal](https://support.astronomer.io).

> - A user can create a new Airflow Deployment via the Astronomer UI, CLI, or API with any supported version of Astronomer Certified.

Reviewer: ```suggestion
- A user can create a new Airflow Deployment via the Astronomer UI, CLI, or API with any supported version of AC.
```

When a version of Airflow is no longer supported, the following is true:

> - The Astronomer Support team is not obligated to answer questions regarding an Airflow Deployment that is running a version of Airflow or Astronomer Certified that is no longer supported.

Reviewer: ```suggestion
- The Astronomer Support team is not obligated to answer questions regarding an Airflow Deployment that is running an outdated version.
```

Author: I think we want to be a bit strong-handed here. I wouldn't say a version is "outdated" - that's more of a subjective term? I think "no longer supported" is accurate but let me know if you have a better suggestion to be clearer. Otherwise considered, "that has reached its end of life date".

Reviewer: Agreed! In that case I would cut down of the length by saying "an unsupported version of Airflow or Astronomer Certified"

> - New Airflow Deployments cannot be created with a version of AC that is no longer supported. Deprecated versions will *not* render as an option in the Deployment creation process from the Astronomer UI, API or CLI.

Reviewer: ```suggestion
- New Airflow Deployments cannot be created with a version of AC that is no longer supported. Deprecated versions will *not* render as an option in the Deployment creation process from the Astronomer UI, API, or CLI.
```

- The latest version of the Astronomer CLI will render a warning if a user pushes a Docker image to Astronomer that corresponds to a version of Airflow that is no longer supported.
- Users can continue to locally test versions of AC that are no longer supported via the Astronomer CLI.
- Service will not be interrupted for Airflow Deployments running a version of AC that is no longer supported.

## Support Schedule

> Per Astronomer Certified's 18-month support lifecycle, the exact support timeline for each published version is defined below.

Reviewer: ```suggestion
The following table contains the exact support timeline for each published version. These timelines are based on Astronomer Certified's 18-month support lifecycle.
```

| AC Version | Release Date   | Support Length | End-of-Support Date |
|------------|----------------|----------------|---------------------|
| 1.10.5     | Sep 4, 2019    | 18 Months      | March 4, 2021       |
| 1.10.7     | Dec 20, 2019   | 18 Months      | June 20, 2021       |
| 1.10.10    | April 23, 2020 | 18 Months      | Oct 23, 2021        |
| 1.10.12    | Sep 30, 2020   | 18 Months      | March 30, 2022      |
| 1.10.14    | Dec 10, 2020   | 18 Months      | June 10, 2022       |
| 1.10.15    | March 19, 2021 | 18 Months      | Sep 19, 2022        |
| 2.0.0      | Dec 17, 2020   | 18 Months      | June 17, 2022       |
| 2.0.2      | April 26, 2021 | 18 Months      | Oct 26, 2022        |
| 2.1.0      | May 21, 2021   | 18 Months      | Nov 21, 2022        |

> | 2.1.1      | July 2, 2021   | 18 Months      | Jan 2, 2023         |

Reviewer 2: same here, add 2.1.3
